app/importers/reading/somerville_mega_reading_importer.rb

```diff
@@ -1,67 +1,90 @@
+    # @syncer = ::RecordSyncer.new(log: @log)
+    @fetcher = options.fetch(:fetcher, GoogleSheetsFetcher.new)
   end
 
   def import
-    streaming_csv = read_or_fetch_csv
-    if streaming_csv.nil?
+    streaming_csvs = read_or_fetch_sheet
+    if streaming_csvs.nil?
       log('Aborting since no CSV found...')
       return
     end
 
-    # TODO do any initialization before the core loop
-    # log('Building student_ids_map...')
-    # @student_ids_map.reset!
-    # log("@student_ids_map built with #{@student_ids_map.size} local_id keys")
-
     log('Starting loop...')
-    reset_counters!
-    streaming_csv.each_with_index do |row, index|
-      import_row(row)
-      log("processed #{index} rows.") if index % 1000 == 0
+    @valid_student_names_count = 0
+    @invalid_student_name_count = 0
+    @valid_data_points_count = 0
+
+    processor = MegaReadingProcessor.new(read_uploaded_by_educator_from_env, @school_year)
+    streaming_csvs.each_with_index do |tab, index|
+      rows, meta = processor.process(tab.tab_csv)
+      log("processed #{index} sheets.")
+      rows.each_with_index {|row, index| import_row(row, index)}
+      valid_student_names_count, valid_data_points_count, invalid_student_name_count = meta.values_at(
+        :valid_student_names_count, :valid_data_points_count, :invalid_student_name_count)
+      @valid_student_names_count += valid_student_names_count
+      @invalid_student_name_count += invalid_student_name_count
+      @valid_data_points_count += valid_data_points_count
+
     end
     log('Done loop.')
-    # TODO log counts on stats
-    # log("@skipped_from_school_filter: #{@skipped_from_school_filter}")
-    # log("@skipped_from_student_local_id_filter: #{@skipped_from_student_local_id_filter}")
-    # log("@skipped_from_empty_sep_fieldd_006: #{@skipped_from_empty_sep_fieldd_006}")
+    log("@valid_student_names_count: #{@valid_student_names_count}")
+    log("@invalid_student_names_count: #{@invalid_student_names_count}")
+    log("@valid_data_points_count: #{@valid_data_points_count}")
 
-    log('Calling #delete_unmarked_records...')
-    @syncer.delete_unmarked_records!(records_within_scope)
-    log("Sync stats: #{@syncer.stats}")
+    # log('Calling #delete_unmarked_records...')
+    # @syncer.delete_unmarked_records!(records_within_scope)
+    # log("Sync stats: #{@syncer.stats}")
     nil
   end
 
   private
-  def reset_counters!
-    raise 'unfinished'
-    # TODO set instance variables to 0
+
+  def read_or_fetch_sheet
+    sheet_id = read_sheet_id_from_env
+    streaming_csvs = @fetcher.get_tabs_from_sheet(sheet_id)
   end
 
-  def read_or_fetch_csv
-    raise 'unfinished'
-    # TODO read from google or mock and return `streaming_csv` that responds
-    # to #each_with_index
+  def read_uploaded_by_educator_from_env
+    educator_login_name = ENV.fetch('READING_IMPORTER_UPLOADED_BY_EDUCATOR_LOGIN_NAME', '')
+    uploaded_by_educator = Educator.find_by_login_name(educator_login_name).id
```

Comment:
If env isn't set, the find call will return nil, and then the call to `#id` will raise.

Check out how `StudentVoiceSurveyImporter` uses `.try(:id)` to guard, so that if the env variable is not set, the more informative error below will raise.